pysmt/logics.py

```diff
@@ -798,7 +806,14 @@ def get_closer_logic(supported_logics, logic):
     res = [l for l in supported_logics if logic <= l]
     if len(res) == 0:
         raise NoLogicAvailableError("Logic %s is not supported" % logic)
-    return min(res)
+    # There might be multiple logics that are close.
+    # Instead of throwing an exception (as we do in most_generic_logic),
+    # we enforce a deterministic answer, as we expect this case to be
+    # fairly common.
+    # Note: we could refine this with more heuristics to identify
+    # which logics are preferable, but making this deterministic makes
+    # it easier to spot cases where specifying the logic might be needed.
+    return sorted(res, key=lambda x:str(x))[0]
```

Comment:
I think this is the culprit why the tests are failing: we need to order the logics according to their expressivity and *then* make a deterministic choice among those logics which are incomparable bottoms of the lattice. The problem we have is that the ordering relation among logics is not a total order...